common/token-module/src/lib.rs

```diff
@@ -88,8 +90,34 @@ pub trait TokenModule: fee_estimator_module::FeeEstimatorModule {
         let _ = self.token_whitelist().swap_remove(&token_id);
     }
 
+    #[endpoint(mintToken)]
+    fn mint_token(&self, token_id: &TokenIdentifier, amount: &BigUint) {
+        let caller = self.blockchain().get_caller();
+        require!(caller == self.multi_transfer_contract_address().get(), "Only MultiTransfer can get tokens");
+        if self.mint_burn_allowed(token_id).get() {
+            self.accumulated_burned_tokens(token_id)
+                .update(|burned| *burned -= amount);
+            self.mint_esdt_token(token_id, amount);
+        }
+
+        self.send().direct_esdt(
+            &self.blockchain().get_caller(),
+            token_id,
+            0,
+            amount,
+        );
+    }
+
     // private
 
+    fn burn_esdt_token(&self, token_id: &TokenIdentifier, amount: &BigUint) {
```

Comment:
Not sure if these 2 helping functions are really needed, as they are one liners.

Reply:
It's better for readability.

Reply:
👍